File: laboratorios/lab03/codigo/AyudaEjercicioDeUVA/GraphAM.java
import java.util.ArrayList;
/**
 * Esta clase es una implementación de un grafo usando matrices de adyacencia
 * 
 * @author Mauricio Toro 
 * @version 1
 */
public class GraphAM extends Graph
{
   private int[][] vertices;
   
   public GraphAM(int size)
   {
       super(size);
       vertices = new int[size][size];
       for (int i = 0; i < size; i++)
       {
              for (int j = 0; j < size; j++)
                vertices[i][j] = -1;
       }
   }
   
   public int[][] getEdges()
   {
       return vertices;
    }
   
   public int getWeight(int source, int destination)
   {
       return vertices[source][destination];
    }
   
   public void addArc(int source, int destination, int weight)
   {
       vertices[source][destination] = weight;
       vertices[destination][source] = weight;
   }
  
   public ArrayList<Integer> getSuccessors(int vertex)
   {
      ArrayList<Integer> successors = new ArrayList<Integer>();
      for (int j = 0; j < size; j++)
      {
        if (vertices[vertex][j] != -1)
            successors.add(j);
      }
      return successors;
            
   }
}

File: talleres/taller11/spoiler/python/GraphAm.py
import numpy as np

class GraphAm:

    def __init__(self, size):
        self.size = size
        self.matriz = np.zeros((size, size))
        
    def __str__(self):
        return f'{self.matriz}'

    def getWeight(self, source, destination):
        return self._____[____][_____]

    def addArc(self, source, destination, weight):
        self.matriz[_____][____] = ____

    def getSuccessors(self, vertex):
        succs = []
        for k, v in enumerate(self.matriz[vertex]):
            if v != ___:
                succs.____(____)
        return succs



ga = GraphAm(3)
ga.addArc(0, 1 , 15)
ga.addArc(0, 2 , 30)
ga.addArc(1, 1 , 7)
print(ga)
print(ga.getSuccessors(0))
print(ga.getWeight(1, 1))